tests/Unit/CssInlinerTest.php

```diff
@@ -3140,6 +3140,90 @@ public function inlineCssNotInDebugModeIgnoresOnlyInvalidExcludedSelector(): voi
         self::assertStringContainsString('<p class="z"></p>', $result);
     }
 
+    /**
+     * @test
+     *
+     * @dataProvider excludedCssRuleDataProvider
+     *
+     * @param string $css
+     * @param array<string> $exludedSelectors
+     * @param string $expectedInlineCss
+     */
+    public function addExcludedCssSelectorCanExcludeCssSelectors(string $css, array $exludedSelectors, string $expectedInlineCss): void
+    {
+        $subject = $this->buildDebugSubject('<html><body><section><div class="green"></div></section></body></html>');
+
+        foreach ($exludedSelectors as $exludedSelector) {
+            $subject->addExcludedCssSelector($exludedSelector);
+        }
+        $subject->inlineCss($css);
+
+        self::assertStringContainsString(
+            '<section><div class="green" style="' . $expectedInlineCss . '"></div></section>',
+            $subject->renderBodyContent()
+        );
+    }
+
+    /**
+     * @return array<array{string, array<string>, string}>
+     */
+    public function excludedCssRuleDataProvider(): array
+    {
+        return [
+            'simple selector' => [
+                '* { margin: 0; } .green { color: green; }', // CSS
+                ['*'], // Exclude
+                'color: green;', // Expected inline CSS
```

Comment:
The coding style checker is configured not to allow comments on the same line as statements.

```suggestion
                // CSS
                '* { margin: 0; } .green { color: green; }',
                // Exclude
                ['*'],
                // Expected inline CSS
                'color: green;',
```